### ISSUE TITLE
Event Hubs trigger v5 with user-provided client utilizing Managed Identity

### ISSUE THREAD

**Issue by IvaskevychYuriy** [Azure/azure-sdk-for-net#21416]
**Question**
Hi, I am trying to implement an Azure Function with Event Hubs trigger using prerelease version v5. But without providing "Connection" parameter at all. I see from the docs that it is possible to add functions settings with either conn string or namespace (to utilize Managed Identity). However, I'm using external configuration so I would like to provide my own client for the trigger to use, but I don't see a way to do so and seeking for advice\plans for this "feature?". 

*Working Service Bus sample*
Here is an example of the end goal working for Service Bus, the trigger itself - no explicit connection:

```csharp
ServiceBusTrigger("TOPIC_NAME", "SUBSCRIPTION_NAME")] MyEntity entity
```

Then in a `FunctionsStartup`:

```csharp
public override void Configure(IFunctionsHostBuilder builder)
{
     ....
     Azure.Core.TokenCredential credentials = ....
     builder.Services.AddAzureClients(cfg =>
     {
           cfg.AddServiceBusClientWithNamespace("BUS_NAMESPACE").WithCredential(credentials);
     });
}
```

*Issue with Event Hubs*
Now similar trigger for Event Hubs:

```csharp
[EventHubTrigger("HUB_NAME")] string[] data
```

and relevant section of `FunctionsStartup`:
```csharp
public override void Configure(IFunctionsHostBuilder builder)
{
     ....
     Azure.Core.TokenCredential credentials = ....
     builder.Services.AddAzureClients(cfg =>
     {
           cfg.AddEventHubConsumerClientWithNamespace("$Default", "HUBS_NAMESPACE", "HUB_NAME").WithCredential(credentials);
     });
}
```

I thought that would work the same way, but I get an exception originating from `EventHubClientFactory.GetEventProcessorHost` line 116 (at the time of writing this):
`InvalidOperationException("No event hub receiver named HUB_NAME");`

Looking through the code of (Microsoft.Azure.WebJobs.Extensions.EventHubs 5.0.0-beta.5) library I see why - if I don't provide connection then I cannot use the trigger. Method of interest is `EventHubTriggerAttributeBindingProvider.TryCreateAsync` where the `listener` gets created by invoking `EventHubClientFactory.GetEventProcessorHost` and `EventHubClientFactory.GetEventHubConsumerClient` (there's the same issue if I follow the logic correctly).

So, the question is am I missing something here or there's no way to do it the intended way? 
If no, then if there are any plans\roadmaps\workarounds (as last resort) for this "feature"?

Thanks in advance!

**Environment:**
 - Name and version of the Library package used: `Microsoft.Azure.WebJobs.Extensions.EventHubs 5.0.0-beta.5`
 - Other related packages: `Azure.Messaging.EventHubs 5.4.1`; `Microsoft.Extensions.Azure 1.1.0-beta.3`; `Microsoft.Azure.WebJobs.Extensions.ServiceBus 4.2.1`; `Microsoft.Extensions.Configuration.AzureAppConfiguration 4.1.0`
 - Hosting platform or OS and .NET runtime version: Windows 10 .NET Core 3.1
 - IDE and version : Visual Studio 16.7.6
 - Azure Functions: C#, v3: `Microsoft.NET.Sdk.Functions 3.0.11`

**Comment by jsquire** [comment]
Thank you for your feedback.  Tagging and routing to the team members best able to assist.

**Comment by JoshLove-msft** [comment]
Hi @IvaskevychYuriy,
Can you let us know what your configuration looks like here (making sure to redact any sensitive information)?

Looking through the code, I see that in the case of Service Bus we fall back to a well known config key of AzureWebJobsServiceBus, but there does not appear to be an equivalent for EventHubs.

**Comment by IvaskevychYuriy** [comment]
Hi @JoshLove-msft,
Sure, so configuration-wise we use Azure App Configuration and pull data from there and you're correct, Service Bus uses 'AzureWebJobsServiceBus' by default from what I can see in the code for `ServiceBusTrigger`. 

But actually, I think I've found an issue, after second day of troubleshooting this, it turns out there was exactly 'AzureWebJobsServiceBus' buried in `launchSettings.json` containing a direct connection string that was not replaced on deployment. That alone was causing a lot of troubles on non dev env, but that's another story.

Basically, I've refactored all that stuff into populating env variables (from `local.settings.json` or deployment pipeline). 
Now both `ServiceBusTrigger` and `EventHubTrigger` contain `Connection = "<config_key>"` and Managed Identity works as expected in both cases now when setting `"<config_key>__fullyQualifiedNamespace": "..."` as per docs . It indeed has nothing to do with client provided via `AddAzureClients()` and the trigger creates its own instance. Side note, also bumped package `Microsoft.Azure.WebJobs.Extensions.ServiceBus` to `5.0.0-beta.3` for this to work.

So, I'm sorry, this was my fault - not an issue. Anyway, much thanks for the help, that actually pointed to the real problem!
This topic can be closed then, thanks again!
